### ISSUE TITLE
GPU Monitoring not showing

### ISSUE THREAD

**Issue by BattlestarNX01** [aristocratos/btop4win#10]
Describe the bug

When launching it just shows the normal btop menu without GPU monitoring even when set to true in both the menu and the config.
Work as expected in version 1.0.1 however not in 1.0.3

**To Reproduce**

Install btop-LHM from scoop or github.
Then launch with admin rights.

**Expected behavior**

Should show GPU monitoring

**Screenshots**

**Info (please complete the following information):**
 - btop4win version: btop4win version: 1.0.3
 - Binary: Release/Scoop
 - Architecture: 64bit
 - Windows Version: 22000.856
 - Terminal used: Windows Terminal
 - Font used: Terminus-NF

**Additional context**

contents of `btop.log`

2022/09/11 (21:53:41) | ===> btop++ v.1.0.3
2022/09/11 (21:53:41) | ERROR: Error during Libre Hardware Monitor parsing: invalid stoi argument

**Comment by Noctis-** [comment]
Yes ... this ... wasn't sure if just me ... also, the CPU temp won't show

![image](https://user-images.githubusercontent.com/6160139/190026960-64ffb0d0-e1a6-47c3-a532-8c944f3ce558.png)

Here are my logs: 

```
2022/09/02 (10:59:17) | ===> btop++ v.1.0.1
2022/09/02 (10:59:17) | ERROR: Exception in runner thread -> Proc:: -> invalid string position

2022/09/09 (09:47:07) | ===> btop++ v.1.0.2
2022/09/09 (09:47:07) | ERROR: Error during Libre Hardware Monitor parsing: invalid stoi argument

2022/09/09 (09:50:41) | ===> btop++ v.1.0.2
2022/09/09 (09:50:41) | ERROR: Error during Libre Hardware Monitor parsing: invalid stoi argument
2022/09/09 (09:53:30) | ERROR: Invalid RGB decimal value: "ba1a84"

2022/09/11 (10:18:29) | ===> btop++ v.1.0.2
2022/09/11 (10:18:29) | ERROR: Error during Libre Hardware Monitor parsing: invalid stoi argument

2022/09/11 (10:32:38) | ===> btop++ v.1.0.2
2022/09/11 (10:32:38) | ERROR: Error during Libre Hardware Monitor parsing: invalid stoi argument

2022/09/14 (09:19:12) | ===> btop++ v.1.0.3
2022/09/14 (09:19:12) | ERROR: Error during Libre Hardware Monitor parsing: invalid stoi argument
```

using a ryzen 3900x, and a rtx 2070 super (nvidia) gfx card.

Here's a screenshot from my rainmeter: 
![image](https://user-images.githubusercontent.com/6160139/190027108-b4dcb7e9-57a0-417e-9a60-b0c4f8901bf6.png)

thought : anything to do with the fact that hwmonitor is running as well ? (no idea, shooting in the dark here)

**Comment by Noctis-** [comment]
For reference, here's same version on my laptop : 

![image](https://user-images.githubusercontent.com/6160139/190027714-1967a2f6-5667-45e6-a7ab-97f904ec3691.png)

can't see any error logs on the laptop either

**Comment by Noctis-** [comment]
for the heck of it, turned off hwinfo , closed btop, and re-launch it ... still no temp nor gpu, so i don't think it's a contention for monitoring resources

**Comment by Kodellion** [comment]
Anyone figure this out yet? I thought it was possibly another app hogging the resources I need to access but it doesn't seem like it, everything else I use can pull the information for the gpu from libre, disabled other monitors, lighting cooling control that monitors the gpu, etc. no luck.

**Comment by ultimal** [comment]
This is still an issue

**Comment by Destined2** [comment]
Hey battlestar,
just hit this on AMD (Ryzen + Radeon), Windows 10, btop4win-LHM 1.0.5 and it's a parsing bug, not a driver issue.

Root cause: OHMR_collect() in src/btop_collect.cpp runs every sensor value through std::stoi()/std::stoll(). AMD sensors report floats like 52.0 (sometimes N/A), so the first one throws invalid_argument → the catch block sets has_OHMR = false, silently disabling all temp/GPU monitoring. Matches your log:

ERROR: Error during Libre Hardware Monitor parsing: invalid stoi argument

Fix: parse tolerantly int first, fall back to double (truncate), otherwise skip:

static long long parse_num(const std::string& s) {
      try { return std::stoll(s); }
      catch (...) {
          try { return (long long)std::llround(std::stod(s)); }
          catch (...) {}
      }
      return -1;   // unparseable -> callers skip the sensor
}

applied at every stoi/stoll call in the parse loop (GPU temp/load/mem, CPU core/package temps, clocks). 
After rebuilding, CPU package + per-core temps and GPU stats all display.

There's an existing PR covering this: #66 (safe stoi/stoll, closed without merging). Should land upstream. it currently kills sensor monitoring for anyone with float / N/A values.
hope this help. (i can share the full rebuild if interested)

**Comment by Noctis-** [comment]
Conversation on pr66 was closed many years ago and it seems it was never
merged?

**Comment by Destined2** [comment]
> Conversation on pr66 was closed many years ago and it seems it was never
> merged?
> […](#)

weird, right? 
i might be wrong but the issue is still there :/
